### PR TITLE
Add paasta metastatus utilization api

### LINKED ISSUE #1339: paasta_itests don't run the api server
Currently, to test the api we are just building a fake request object and calling the function in the api.  We should actually run the api server to better simulate the real world.

### PR DESCRIPTION
**Pull request by matthewbentley**
Fixes #1339

### CODE REVIEW

File: paasta_tools/metrics/metastatus_lib.py [hunk 1]
```diff
@@ -377,18 +389,44 @@ def filter_tasks_for_slaves(slaves, tasks):
     return [task for task in tasks if task['slave_id'] in slave_ids]
 
 
-def get_resource_utilization_by_grouping(grouping_func, mesos_state):
+def filter_slaves(slaves, filters):
```
Comment (Rob-Johnson) [inline paasta_tools/metrics/metastatus_lib.py]: can you make this more generic? make filters be a list of predicate fns that match on the slave object and do something like:

```
return [s for s in slaves if all[f(s) for f in filters]]
```

File: paasta_tools/api/views/resources.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    #  "pattern": "(.*):(.*,)*(.*)"
+    if filters is None:
+        return {}
+    f = {s[0]: s[1] for s in [e.split(':') for e in filters]}
+    f = {k: v.split(',') for k, v in f.items()}
+    return f
+
+
+@view_config(route_name='resources.utilization', request_method='GET', renderer='json')
+def resourses_utilization(request):
```
Comment (Rob-Johnson) [inline paasta_tools/api/views/resources.py]: resources

File: paasta_tools/api/views/resources.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    master = get_mesos_master()
+    mesos_state = master.state
+
+    groupings = request.swagger_data.get('groupings', [])
+    filters = request.swagger_data.get('filter', [])
+    filters = parse_filters(filters)
+    grouping_function = metastatus_lib.key_func_for_attribute_multi(groupings)
+    resource_info_dict = metastatus_lib.get_resource_utilization_by_grouping(grouping_function,
```
Comment (Rob-Johnson) [inline paasta_tools/api/views/resources.py]: kwargs this please

**Comment by Rob-Johnson** [comment]
unit tests + itests please

**Comment by matthewbentley** [comment]
At this point most of this PR is tests.  It re-works the api itests to use an actual server and runs some basic sanity checks on the api.  However, it will be difficult to write more robust itests for the resources api without adding code to manipulate the running mesosslaves.  Which should be possible, just rather involved.

**Comment by matthewbentley** [comment]
@Rob-Johnson can you look at this again?

File: paasta_itests/steps/paasta_api_steps.py [hunk 4]
```diff
@@ -15,40 +15,55 @@
...
+            service=service
+        ).result()
+    except bexception.HTTPError as exc:
+        assert exc.status_code == int(error_code)
 
     assert not response
+
+
+@then('resources GET should show "{resource}" has {used:d} used')
+def resouces_resource_used(context, resource, used):
```
Comment (Rob-Johnson) [inline paasta_itests/steps/paasta_api_steps.py]: s/resouces/resources

File: paasta_itests/steps/paasta_api_steps.py [hunk 5]
```diff
@@ -15,40 +15,55 @@
...
+    groupings = groupings.split(",")
+    if len(filters) > 0:
+        filters = filters.split("|")
+    response = context.paasta_api_client.resources.resources(groupings=groupings, filter=filters).result()
+
+    assert len(response) == num, response
+
+
+@then('resources GET with groupings "{groupings}" should return {num:d} groups')
+def resourses_groupings(context, groupings, num):
```
Comment (Rob-Johnson) [inline paasta_itests/steps/paasta_api_steps.py]: s/resourses/resources

**Pull request merged**